repo/format/upgrade_lock_test.go

```diff
@@ -116,6 +117,77 @@ func TestFormatUpgradeCommit(t *testing.T) {
 	require.EqualError(t, env.RepositoryWriter.FormatManager().RollbackUpgrade(ctx), "no upgrade in progress")
 }
 
+func TestFormatUpgradeValidate(t *testing.T) {
```

Comment:
might be a good idea to simulate a negative test-case as well to ensure upgrade fails if validate fails ..

Since all the blobs are in-memory, you can access the blob-storage and corrupt it in a way you would get a report of length more than zero.

Reply:
@Shrekster agreed.  Unsure as of yet how that happens.  Likely to depend on the final implementation of the index checker.